code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -81,6 +81,7 @@
  * * added_ph - override to force a pH when added
  * * override_base_ph - ingore the present pH of the reagent, and instead use the default (i.e. if buffers/reactions alter it)
  * * ignore splitting - Don't call the process that handles reagent spliting in a mob (impure/inverse) - generally leave this false unless you care about REAGENTS_DONOTSPLIT flags (see reagent defines)
+ * * list/reagent_added - If not null will contain the reagent datum that was added after this proc returns. Clear the list to erase old values
```

Comment:
This 100% needs documentation explaining what it's used for, because I cannot remotely understand why we would be doing this 

`add_reagent` *should* be simple to use since so many people interface with it so the more we complicate it the worse it gets

Reply:
Here's the explanation from https://github.com/tgstation/tgstation/pull/90930#issuecomment-2847385150

> Thing is add_reagent() already returns a number that indicates the volume of reagent added but we also need it to return the actual reagent datum it created(did it create a inverse or the original?)but it can't do that cause a proc can return only 1 value so to get that reagent datum it created we pass a list holder as a param that will store that datum.
> That list is used by proc trans_to().

Reply:
> add_reagent should be simple to use since so many people interface with it so the more we complicate it the worse it gets

Rest assured. This is an optional arg so it  can be null all the time. To better understand what this arg does here's an example on how you would use it.

Say you want to add a reagent and get that reagent datum you added all in one line. Here's how it's done

```
var/list/reagent_datum = list()
var/amount_added= add_reagent(/datum/reagent/water, 5, var/datum/reagent/water/reagent_added = reagent_datum[1]
```

Now we don't need to use has_reagent() to retrieve that reagent we just added